src/fsharp/FSharp.Compiler.Service.ProjectCracker.Exe/Program.fs

```diff
@@ -378,20 +378,24 @@ module Program =
       // C# referenced projects, as we don't look at them otherwise.
       let referencedProjectOutputs =
           if runningOnMono then
-              [| yield! Array.map (fun (s,_) -> "-r:" + s) referencedProjectOptions
-                 for file in parsedProject.ProjectReferences do
-                     let ext = Path.GetExtension(file)
-                     if ext = ".csproj" || ext = ".vbproj" then
-                         let parsedProject = FSharpProjectFileInfo.Parse(file, properties=properties, enableLogging=false)
-                         match parsedProject.OutputFile with
-                         | None -> ()
-                         | Some f -> yield "-r:" + f |]
+              [ yield! Array.map (fun (s,_) -> "-r:" + s) referencedProjectOptions
+                for file in parsedProject.ProjectReferences do
+                    let ext = Path.GetExtension(file)
+                    if ext = ".csproj" || ext = ".vbproj" then
+                        let parsedProject = FSharpProjectFileInfo.Parse(file, properties=properties, enableLogging=false)
+                        match parsedProject.OutputFile with
+                        | None -> ()
+                        | Some f -> yield "-r:" + f ]
           else
-              [||]
+              []
+
+      // On some versions of Mono the referenced projects are already
+      // correctly included, so we make sure there aren't any duplicates
+      let cmdOptions = Array.ofSeq (Set.union (set parsedProject.Options)
```

Comment:
Hmm..  I'd be more comfortable if 

1. keep the options in the same order and 

2. we only remove duplicates for ``-r`` options.  I'm concerned that sequences like ``--optimize-`` ``--optimize+`` ``--optimize-`` will just become ``--optimize-`` ``--optimize+`` - I'm not sure this can really happen but it's best to avoid the possibility.